### PR TITLE
Synonyms v2

### LINKED ISSUE #23: Synonym v2 integration
See: https://github.com/algolia/algoliasearch-client-csharp/compare/master...newsynonyms

### PR DESCRIPTION
**Pull request by ElPicador**
Closes #23

### CODE REVIEW

**Comment by ElPicador** [comment]
@redox @Nagriar ready to review

**Comment by PLNech** [comment]
LGTM :+1:

**Comment by ElPicador** [comment]
Waiting for full deploy to merge

**Comment by Nagriar** [comment]
@ElPicador We can merge it.

File: src/main/java/com/algolia/search/saas/Index.java [hunk 1]
```diff
@@ -960,4 +961,130 @@ public JSONObject searchDisjunctiveFaceting(Query query, List<String> disjunctiv
...
+
+    /**
+     * Update one synonym
+     *
+     * @param objectID                  The objectId of the synonym to save
+     * @param content                   The new content of this synonym
+     * @param forwardToSlaves           Forward the operation to the slave indices
+     * @param replaceExistingSynonyms   Replace the existing synonyms with this save
+     */
+    public JSONObject saveSynonym(String objectID, JSONObject content, boolean forwardToSlaves, boolean replaceExistingSynonyms) throws AlgoliaException {
```
Comment (olance) [inline src/main/java/com/algolia/search/saas/Index.java:1076]: There's no `replaceExistingSynonyms` param for the single synonym PUT request. Just `forwardToSlaves`.

**Comment by ElPicador** [comment]
Adding a new issue to fix this: #37

**Pull request merged**